### PR TITLE
Added difficulty drop down to the chart editor

### PR DESCRIPTION
**Pull request by CerBor**
This PR adds ability to open song difficulties files in charting editor.

### CODE REVIEW

**Comment by Gui-iago** [comment]
it crashes if you open it on Masters Menu

File: source/editors/ChartingState.hx [hunk 1]
```diff
@@ -576,16 +576,21 @@ class ChartingState extends MusicBeatState
...
+			if (newDifficulty != 'normal') {
 				postfix = '-' + newDifficulty;
+			} else {
+				postfix = '';
+			}
+			try {
+				PlayState.SONG = Song.loadFromJson(_song.song.toLowerCase() + postfix, _song.song.toLowerCase());
+				MusicBeatState.resetState();
+			} catch (e:Any) {
+				trace("File " + _song.song.toLowerCase() + postfix + " is not found.");
```
Comment (Gui-iago) [inline source/editors/ChartingState.hx:592]: image
L O N G

**Comment by CerBor** [comment]
for me it all seems working normal

**Comment by Gui-iago** [comment]
> for me it all seems working normal

it was a problem with my build, found a way to fix it already, hoping for this to be merged

**Comment by CerBor** [comment]
thanks :)

**Comment by CerBor** [comment]
Can someone merge this? @bbpanzu

**Comment by Gui-iago** [comment]
@bbpanzu ayo?

**Comment by CerBor** [comment]
@bbpanzu?

**Comment by JorgeMiguel22** [comment]
source/editors/ChartingState.hx:219: characters 9-28 : Class<Paths> has no field destroyLoadedImages


yeahhhhh another error while building

**Comment by MisterRafers** [comment]
You need to compile code from my branch because i have older version of engine.

**Comment by MisterRafers** [comment]
I mean, all files from branch

**Comment by CerBor** [comment]
You need to compile all files from my branch because i have older version of engine. Or just copy paste my changes in your code

**Comment by GrexpungedYT** [comment]
795CAEC0-C282-47B0-BE02-26F2BCE833C6
fix it pls

**Review by Raltyro — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by TheKitBoi — APPROVED** [review]

**Pull request closed without merging**